src/main/java/org/assertj/core/internal/Uris.java

```diff
@@ -81,4 +89,91 @@ public void assertHasUserInfo(AssertionInfo info, URI actual, String expected) {
     assertNotNull(info, actual);
     if (!areEqual(actual.getUserInfo(), expected)) throw failures.failure(info, shouldHaveUserInfo(actual, expected));
   }
+
+  @VisibleForTesting
+  public static Map<String, List<String>> getParameters(String query) {
+    Map<String, List<String>> parameters = new HashMap<String, List<String>>();
+
+    if (query != null && !query.isEmpty()) {
+      for (String pair : query.split("&")) {
+        int equalSign = pair.indexOf("=");
+        String key = equalSign == -1 ? pair : pair.substring(0, equalSign);
+        String value = equalSign == -1 ? null : pair.substring(equalSign + 1);
+
+        try {
+          key = URLDecoder.decode(key, "UTF-8");
+        } catch (UnsupportedEncodingException ex) {
+          // UTF-8 is missing? Allow the key to remain encoded (no reasonable alternative).
+        }
+
+        if (value != null) {
+          try {
+            value = URLDecoder.decode(value, "UTF-8");
+          } catch (UnsupportedEncodingException ex) {
+            // UTF-8 is missing? Allow the value to remain encoded (no reasonable alternative).
+          }
+        }
+
+        if (!parameters.containsKey(key)) {
+          parameters.put(key, new ArrayList<String>());
+        }
+
+        parameters.get(key).add(value);
+      }
+    }
+
+    return parameters;
+  }
+
+  public void assertHasParameter(AssertionInfo info, URI actual, String name) {
+    assertNotNull(info, actual);
+
+    Map<String, List<String>> parameters = getParameters(actual.getQuery());
+
+    if (!parameters.containsKey(name)) {
+      throw failures.failure(info, shouldHaveParameter(actual, name));
+    }
+  }
+
+  public void assertHasParameter(AssertionInfo info, URI actual, String name, String value) {
+    assertNotNull(info, actual);
+
+    Map<String, List<String>> parameters = getParameters(actual.getQuery());
+
+    if (!parameters.containsKey(name)) {
+      throw failures.failure(info, shouldHaveParameter(actual, name, value));
+    }
+
+    List<String> values = parameters.get(name);
+
+    if (!values.contains(value)) {
+      throw failures.failure(info, shouldHaveParameter(actual, name, value, values.get(0)));
```

Comment:
are we sure values can't be empty ?
If we display parameter's values, we should display them all between `[]`, if there is only one, stick with the actual message.